x-pack/plugins/data_enhanced/public/search/search_interceptor.ts

```diff
@@ -91,18 +122,94 @@ export class EnhancedSearchInterceptor extends SearchInterceptor {
       tap((response) => (id = response.id)),
       catchError((e: Error) => {
         cancel();
-        return throwError(this.handleSearchError(e, options, searchAbortController.isTimeout()));
+        return throwError(e);
       }),
       finalize(() => {
-        this.pendingCount$.next(this.pendingCount$.getValue() - 1);
         searchAbortController.cleanup();
-        if (untrackSearch && this.deps.session.isCurrentSession(options.sessionId)) {
-          // untrack if this search still belongs to current session
-          untrackSearch();
-        }
         if (savedToBackgroundSub) {
           savedToBackgroundSub.unsubscribe();
         }
+      }),
+      shareReplay(1)
+    );
+  }
+
+  /**
+   * @internal
+   * Creates a new search observable and a corresponding search abort controller
+   * If requestHash is defined, tries to return them first from cache.
+   */
+  private getSearchResponse$(
+    request: IKibanaSearchRequest,
+    options: IAsyncSearchOptions,
+    requestHash?: string
+  ) {
+    const cached = requestHash ? this.responseCache.get(requestHash) : undefined;
+
+    const searchAbortController =
+      cached?.searchAbortController || new SearchAbortController(this.searchTimeout);
+
+    // Create a new abort signal if one was not passed. This fake signal will never be aborted,
+    // So the underlaying search will not be aborted, even if the other consumers abort.
+    searchAbortController.addAbortSignal(options.abortSignal || new AbortController().signal);
+    const response$ = cached?.response$ || this.runSearch$(request, options, searchAbortController);
+
+    if (requestHash && !this.responseCache.has(requestHash)) {
+      this.responseCache.set(requestHash, {
+        response$,
+        searchAbortController,
+      });
+    }
+
+    return {
+      response$,
+      searchAbortController,
+    };
+  }
+
+  public search({ id, ...request }: IKibanaSearchRequest, options: IAsyncSearchOptions = {}) {
+    const searchOptions = {
+      strategy: ENHANCED_ES_SEARCH_STRATEGY,
+      ...options,
+    };
+    const { sessionId, abortSignal } = searchOptions;
+
+    return this.createRequestHash$(request, options).pipe(
```

Comment:
Here and everywhere below where we pass `options`, I think we should use `searchOptions` otherwise `strategy` might be undefined. 
And I found an edge case where it breaks things a bit (not sure if it is a regression of this pr):

Screenshot when stoping a search session: 
![Screen Shot 2021-04-07 at 15 13 08](https://user-images.githubusercontent.com/7784120/113872083-d5fd0080-97b3-11eb-8308-70713de3cf8e.png)

Reply:
See `undefined` inside the URL